src/solids4FoamModels/materialModels/mechanicalModel/mechanicalLaws/linearGeometryLaws/linearElasticMohrCoulombPlastic/linearElasticMohrCoulombPlastic.C

```diff
@@ -942,5 +944,68 @@ void Foam::linearElasticMohrCoulombPlastic::updateTotalFields()
         << nl << endl;
 }
 
+void Foam::linearElasticMohrCoulombPlastic::writeFields(const Time& runTime)
+{
+    if(mesh()!=baseMesh()) // possible: lawI < 0, not sure about dualMesh
+    {
+        const solidSubMeshes& solSubMeshesRef = baseMesh().lookupObject<mechanicalModel>
+                (
+                   "mechanicalProperties"
+                ).solSubMeshes();
+        PtrList<volScalarField> subMeshYields(solSubMeshesRef.subMeshes().size());
+        IOobject defaultIO
+        (
+            "subMeshYield",
+            runTime.timeName(),
+            mesh(),
+            IOobject::NO_READ,
+            IOobject::NO_WRITE,
+            false
+        );
+        dimensionedScalar zero("",dimless,0.0);
+        forAll(solSubMeshesRef.subMeshes(),iLaw)
```

Comment:
@denMaier: does this procedure make sense if there is more than one linearElasticMohrCoulomb law in the case? e.g. two materials, where both are linearElasticMohrCoulomb but with different parameters. I think it still works, as the field is named `Yield.<name>` where `<name>` is the unique material name in mechanicalProperties.